flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointFailureManager.java

```diff
@@ -170,22 +233,7 @@ private void clearCount() {
         countedCheckpointIds.clear();
     }
 
-    /**
-     * Fails the whole job graph in case an in-progress synchronous savepoint is discarded.
-     *
-     * <p>If the checkpoint was cancelled at the checkpoint coordinator, i.e. before the synchronous
-     * savepoint barrier was sent to the tasks, then we do not cancel the job as we do not risk
-     * having a deadlock.
-     *
-     * @param cause The reason why the job is cancelled.
-     */
-    void handleSynchronousSavepointFailure(final Throwable cause) {
-        if (!isPreFlightFailure(cause)) {
-            failureCallback.failJob(cause);
-        }
-    }
-
-    private static boolean isPreFlightFailure(final Throwable cause) {
+    private static boolean isJMThrowable(final Throwable cause) {
```

Comment:
`isPreFlightFailure` in this case I think makes more sense for the sake of the consistency with `CheckpointFailureReason#isPreFlight`. At least as long as we want to keep `CheckpointFailureReason#isPreFlight` name, as I think it might be a good idea to rename it to `isJobManagerFailure()`.